### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by plankton9991**
[DEMO LINK](https://plankton9991.github.io/react_person/)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -28,22 +29,29 @@ const alex = {
...
-      <p className="Person__age">I am 37</p>
-      <p className="Person__partner">My wife&apos;s name is Natasha</p>
-    </section>
+    <Person
+      name={misha.name}
+      age={misha.age}
+      sex={misha.sex}
+      isMarried={misha.isMarried}
+      partnerName={misha.partnerName}
+    />
```
Comment (misha200119) [inline src/App.js:38]: better set dataObject as 1 prop and destructure it in Person component

File: src/App.js [hunk 2]
```diff
@@ -28,22 +29,29 @@ const alex = {
...
-      <h2 className="Person__name">My name is Olya</h2>
-      <p className="Person__partner">My husband&apos;s name is Maksym</p>
-    </section>
+    <Person
+      name={olya.name}
+      age={olya.age}
+      sex={olya.sex}
+      isMarried={olya.isMarried}
+      partnerName={olya.partnerName}
+    />
```
Comment (misha200119) [inline src/App.js:46]: same

**Review by misha200119 — APPROVED** [review]

File: src/App.js [hunk 3]
```diff
@@ -28,22 +29,29 @@ const alex = {
...
-      <h2 className="Person__name">My name is Misha</h2>
-      <p className="Person__age">I am 37</p>
-      <p className="Person__partner">My wife&apos;s name is Natasha</p>
-    </section>
+    <Person
+      name={misha.name}
+      age={misha.age}
+      sex={misha.sex}
+      isMarried={misha.isMarried}
+      partnerName={misha.partnerName}
```
Comment (MaksimusCRS) [inline src/App.js:37]:
```suggestion
      person={misha}
```

File: src/App.js [hunk 4]
```diff
@@ -28,22 +29,29 @@ const alex = {
...
-    <section className="Person">
-      <h2 className="Person__name">My name is Olya</h2>
-      <p className="Person__partner">My husband&apos;s name is Maksym</p>
-    </section>
+    <Person
+      name={olya.name}
+      age={olya.age}
+      sex={olya.sex}
+      isMarried={olya.isMarried}
+      partnerName={olya.partnerName}
```
Comment (MaksimusCRS) [inline src/App.js:45]: same

File: src/App.js [hunk 5]
```diff
@@ -28,22 +29,29 @@ const alex = {
...
-      <h2 className="Person__name">My name is Alex</h2>
-      <p className="Person__age">I am 25</p>
-      <p className="Person__partner">I am not married</p>
-    </section>
+    <Person
+      name={alex.name}
+      age={alex.age}
+      sex={alex.sex}
+      isMarried={alex.isMarried}
+      partnerName={alex.partnerName}
```
Comment (MaksimusCRS) [inline src/App.js:53]: same

File: src/components/Person/Person.jsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  name,
+  age,
+  sex,
+  isMarried,
+  partnerName,
+}) => (
+  <section className="Person">
+    <h2 className="Person__name">{`My name is ${name}`}</h2>
+    {age && (
+      <p className="Person__age">{`I am ${age}`}</p>
```
Comment (MaksimusCRS) [inline src/components/Person/Person.jsx:15]:
```suggestion
      <p className="Person__age">
        {`I am ${age}`}
      </p>
```

File: src/components/Person/Person.jsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    )}
+    {isMarried ? (
+      <p className="Person__partner">
+        {`My ${sex === 'm'
+          ? "wife'"
+          : "husband'"}s name is ${partnerName}`}
```
Comment (MaksimusCRS) [inline src/components/Person/Person.jsx:21]: You need variable for sex === 'm'
          ? "wife'"
          : "husband

**Review by MaksimusCRS — CHANGES_REQUESTED** [review]

File: src/components/Person/Person.jsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        <p className="Person__age">
+          {`I am ${age}`}
+        </p>
+      )}
+      {isMarried ? (
+        <p className="Person__partner">
+          {`My ${partner}s name is ${partnerName}`}
+        </p>
+      ) : (
+        <p className="Person__partner">I am not married</p>
```
Comment (MaksimusCRS) [inline src/components/Person/Person.jsx:31]: you just change text here

**Review by MaksimusCRS — APPROVED** [review]

**Pull request closed without merging**